trustfall_core/src/interpreter/hints/vertex_info.rs

```diff
@@ -132,6 +146,57 @@ impl<T: InternalVertexInfo + super::sealed::__Sealed> VertexInfo for T {
         }
     }
 
+    /// required_properties returns an iterator over all properties needed
+    /// for this vertex.
+    fn required_properties(&self) -> Vec<RequiredProperty> {
+        let current_component = self.current_component();
+
+        let current_vertex = self.current_vertex();
+
+        // we need all output properties
+        let mut properties: Vec<RequiredProperty> = current_component
+            .outputs
+            .values()
+            .map(|c| RequiredProperty::new(c.field_name.clone()))
+            .collect::<Vec<RequiredProperty>>();
+
+        // extend with the edges from this vertex
+        properties.extend(
+            current_component
+                .edges
+                .values()
+                .map(|a| RequiredProperty::new(a.edge_name.clone()))
+                .collect::<Vec<RequiredProperty>>(),
+        );
+
+        properties.extend(
+            current_component
+                .folds
+                .values()
+                .map(|i| RequiredProperty::new(i.edge_name.clone()))
+                .collect::<Vec<RequiredProperty>>(),
+        );
```

Comment:
Edges != properties, those are two disjoint sets of fields. I believe there's already a function for edges in the API.